### ISSUE TITLE
Overfitting Reranker model

### ISSUE THREAD

**Issue by dohuyduc2002** [FlagOpen/FlagEmbedding#1272]
I have tuned bge-reranker-v2 with these hyperparameters. Even I use the reranker for Vietnamese MTEB results, it's seems that reranker model degraded the quality of my previous tuned embedding model which is BGE-m3. In theory, to achieve good-quality results for embedding, reranker should be added after the embedded vector. 

At the first tuning, I used the following configuration:
- Learning Rate: `4e-6` (no scheduler and decay)
- The model converged well, but smaller learning rates led to overfitting.

Below is the full parameter configuration:

```bash
--learning_rate 4e-6 \
--fp16 True \
--num_train_epochs 3 \
--per_device_train_batch_size 4 \
--gradient_accumulation_steps 8 \
--dataloader_drop_last True \
--train_group_size 2 \
--max_len 512 \
--weight_decay 0.01 \
--logging_steps 100 \
--warmup_ratio 0.1 \
--warmup_steps 2000 \
--lr_scheduler_type 'cosine' \
--save_strategy "epoch"

![Screenshot 2024-12-02 at 18 42 36](https://github.com/user-attachments/assets/6a0b6d03-1bf0-4cee-b388-8370759835bc)

**Comment by 545999961** [comment]
You can try setting `train_group_size` to 16 and fine-tune with scheduler.
